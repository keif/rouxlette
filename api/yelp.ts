import axios from "axios";
import { YELP_API_KEY } from "@env";
import { logSafe } from "../utils/log";

const YELP_URL = `https://api.yelp.com/v3`;

const yelp = axios.create({
	baseURL: YELP_URL,
	headers: {
		Authorization: `Bearer ${YELP_API_KEY}`,
	},
});

yelp.interceptors.request.use(function (config) {
	logSafe('YELP REQUEST', {
		method: config.method?.toUpperCase(),
		url: config.url, // Just the path, not full URL to avoid redundancy
		params: config.params ? Object.keys(config.params) : [],
		hasAuth: !!config.headers?.Authorization
	});
	return config;
}, function (error) {
	logSafe('YELP REQUEST ERROR', { message: error?.message });
	return Promise.reject(error);
});

yelp.interceptors.response.use(function (response) {
	logSafe('YELP RESPONSE', {
		status: response.status,
		url: response.config?.url,
		businessCount: response.data?.businesses?.length || 0,
		totalResults: response.data?.total || 0
	});
	return response;
}, function (error) {
	logSafe('YELP RESPONSE ERROR', {
		status: error?.response?.status,
		message: error?.message,
		url: error?.config?.url
	});
	return Promise.reject(error);
});

/**
 * Get detailed business information including hours, photos, and phone
 * @param id - The Yelp business ID
 * @returns Promise<YelpBusiness> - Full business details
 */
export async function getBusinessDetails(id: string) {
  const response = await yelp.get(`/businesses/${id}`);
  return response.data;
}

export default yelp;
