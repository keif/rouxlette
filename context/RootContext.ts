import { createContext, Dispatch } from "react";
import { AppState, initialAppState } from "./state";
import { AppActions } from "./actions";

export const RootContext = createContext<{
	state: AppState;
	dispatch: Dispatch<AppActions>;
}>({
	state: initialAppState,
	dispatch: () => undefined,
});
