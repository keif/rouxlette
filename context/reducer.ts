import { AppState, Filter } from "./state";
import {
	ActionType,
	AppActions,
	SetCategories,
	SetDetail,
	SetFilter,
	SetLocation,
	SetResults,
	SetShowFilter,
} from "./actions";
import { Category, Result } from "../hooks/useResults";

export function appReducer(state: AppState, action: AppActions): AppState {
	switch (action.type) {
		case ActionType.SetCategories:
			return {
				...state,
				categories: action.payload.categories,
			};
		case ActionType.SetDetail:
			return {
				...state,
				detail: action.payload.detail,
			};
		case ActionType.SetFilter:
			return {
				...state,
				filter: action.payload.filter,
			};
		case ActionType.SetLocation:
			return {
				...state,
				location: action.payload.location,
			};
		case ActionType.SetResults:
			return {
				...state,
				results: action.payload.results,
			};
		case ActionType.SetShowFilter:
			return {
				...state,
				showFilter: action.payload.showFilter,
			};
		default:
			return state;
	}
}

// helper functions to simplify the caller
export const setCategories = (categories: Category[]): SetCategories => ({
	type: ActionType.SetCategories,
	payload: { categories },
});

export const setDetail = (detail: Result): SetDetail => ({
	type: ActionType.SetDetail,
	payload: { detail },
});

export const setFilter = (filter: Filter): SetFilter => ({
	type: ActionType.SetFilter,
	payload: { filter },
});
export const setLocation = (location: string): SetLocation => ({
	type: ActionType.SetLocation,
	payload: { location },
});

export const setResults = (results: Result[]): SetResults => ({
	type: ActionType.SetResults,
	payload: { results },
});

export const setShowFilter = (showFilter: boolean): SetShowFilter => ({
	type: ActionType.SetShowFilter,
	payload: { showFilter },
});
