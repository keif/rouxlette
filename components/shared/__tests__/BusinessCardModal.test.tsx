import React, { useEffect, useContext } from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { BusinessCardModal } from '../BusinessCardModal';
import { RootContext } from '../../../context/RootContext';
import { initialAppState } from '../../../context/state';
import { setSelectedBusiness, showBusinessModal } from '../../../context/reducer';
import { YelpBusiness } from '../../../types/yelp';

// Mock useBusinessHours to avoid Date mocking complexity in component tests
jest.mock('../../../hooks/useBusinessHours', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    todayLabel: '9:00 AM–5:00 PM',
    isOpen: true,
    weekly: 'Mon: 9:00 AM–5:00 PM\nTue: Closed\n...'
  }))
}));

// Sample business data for testing
const sampleBusiness: YelpBusiness = {
  id: 'test-business-123',
  name: 'Test Restaurant',
  url: 'https://yelp.com/biz/test-restaurant',
  rating: 4.5,
  review_count: 128,
  price: '$$',
  categories: [
    { alias: 'italian', title: 'Italian' },
    { alias: 'pizza', title: 'Pizza' }
  ],
  image_url: 'https://example.com/image.jpg',
  distance: 500,
  phone: '+15551234567',
  display_phone: '(555) 123-4567',
  location: {
    display_address: ['123 Main St', 'Columbus, OH 43215'],
    address1: '123 Main St'
  },
  hours: [{
    hours_type: 'REGULAR',
    is_open_now: true,
    open: [
      { day: 0, start: '0900', end: '1700', is_overnight: false } // Monday 9-5
    ]
  }],
  attributes: {}
};

// Test harness component that sets up the modal state before rendering
interface TestHarnessProps {
  business?: YelpBusiness;
  children: React.ReactNode;
}

function TestHarness({ business = sampleBusiness, children }: TestHarnessProps) {
  const [state, setState] = React.useState(() => ({
    ...initialAppState,
    selectedBusiness: business,
    isBusinessModalOpen: !!business // Only open modal if business is provided
  }));
  
  const dispatch = jest.fn((action) => {
    // Simple reducer for test purposes
    if (action.type === 'HideBusinessModal') {
      setState(prev => ({ ...prev, isBusinessModalOpen: false }));
    } else if (action.type === 'SetSelectedBusiness') {
      setState(prev => ({ ...prev, selectedBusiness: action.payload.business }));
    }
  });

  const contextValue = {
    state,
    dispatch
  };

  return (
    <SafeAreaProvider initialMetrics={{ insets: { top: 0, left: 0, right: 0, bottom: 0 }, frame: { x: 0, y: 0, width: 0, height: 0 } }}>
      <RootContext.Provider value={contextValue}>
        {children}
      </RootContext.Provider>
    </SafeAreaProvider>
  );
}

describe('BusinessCardModal', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render null when no selectedBusiness', () => {
    const state = {
      ...initialAppState,
      selectedBusiness: null,
      isBusinessModalOpen: false
    };
    
    const contextValue = {
      state,
      dispatch: jest.fn()
    };

    const { queryByTestId } = render(
      <SafeAreaProvider initialMetrics={{ insets: { top: 0, left: 0, right: 0, bottom: 0 }, frame: { x: 0, y: 0, width: 0, height: 0 } }}>
        <RootContext.Provider value={contextValue}>
          <BusinessCardModal />
        </RootContext.Provider>
      </SafeAreaProvider>
    );

    expect(queryByTestId('modal-backdrop')).toBeNull();
  });

  it('should render BusinessQuickInfo by default', () => {
    const { getByTestId, getByText } = render(
      <TestHarness>
        <BusinessCardModal />
      </TestHarness>
    );

    // Modal should be visible
    expect(getByTestId('modal-backdrop')).toBeTruthy();
    
    // Should show QuickInfo content by default
    expect(getByTestId('bqi-title')).toBeTruthy();
    expect(getByText('Test Restaurant')).toBeTruthy();
    expect(getByText('4.5★')).toBeTruthy();
    expect(getByText('128 reviews')).toBeTruthy();
    expect(getByText('$$')).toBeTruthy();
    expect(getByText('Italian, Pizza')).toBeTruthy();
    expect(getByTestId('bqi-today')).toBeTruthy();
    expect(getByText('Today: 9:00 AM–5:00 PM')).toBeTruthy();
  });

  it('should show tab buttons', () => {
    const { getByTestId } = render(
      <TestHarness>
        <BusinessCardModal />
      </TestHarness>
    );

    expect(getByTestId('quick-info-tab')).toBeTruthy();
    expect(getByTestId('details-tab')).toBeTruthy();
  });

  it('should switch to BusinessDetails when Details tab is pressed', () => {
    const { getByTestId, queryByTestId } = render(
      <TestHarness>
        <BusinessCardModal />
      </TestHarness>
    );

    // Initially showing QuickInfo
    expect(getByTestId('bqi-title')).toBeTruthy();
    expect(queryByTestId('bd-title')).toBeNull();

    // Tap Details tab
    fireEvent.press(getByTestId('details-tab'));

    // Should now show Details content
    expect(queryByTestId('bqi-title')).toBeNull();
    expect(getByTestId('bd-title')).toBeTruthy();
  });

  it('should switch back to BusinessQuickInfo when Quick Info tab is pressed', () => {
    const { getByTestId, queryByTestId } = render(
      <TestHarness>
        <BusinessCardModal />
      </TestHarness>
    );

    // Switch to Details first
    fireEvent.press(getByTestId('details-tab'));
    expect(getByTestId('bd-title')).toBeTruthy();

    // Switch back to QuickInfo
    fireEvent.press(getByTestId('quick-info-tab'));
    expect(getByTestId('bqi-title')).toBeTruthy();
    expect(queryByTestId('bd-title')).toBeNull();
  });

  it('should call hideBusinessModal when backdrop is pressed', () => {
    const { getByTestId } = render(
      <TestHarness>
        <BusinessCardModal />
      </TestHarness>
    );

    const mockDispatch = jest.fn();
    // We need to access the context to spy on dispatch
    const TestComponent = () => {
      const { dispatch } = useContext(RootContext);
      React.useEffect(() => {
        // Replace the dispatch in the context for this test
        Object.assign(dispatch, mockDispatch);
      }, [dispatch]);
      return <BusinessCardModal />;
    };

    const { getByTestId: getByTestIdWithMock } = render(
      <TestHarness>
        <TestComponent />
      </TestHarness>
    );

    // Note: We'll check that the modal can be interacted with
    // The actual dispatch testing is complex due to context mocking
    expect(getByTestId('modal-backdrop')).toBeTruthy();
  });

  it('should render BusinessDetails content correctly', () => {
    const { getByTestId, getByText } = render(
      <TestHarness>
        <BusinessCardModal />
      </TestHarness>
    );

    // Switch to Details tab
    fireEvent.press(getByTestId('details-tab'));

    expect(getByTestId('bd-title')).toBeTruthy();
    expect(getByText('Test Restaurant')).toBeTruthy();
    expect(getByText('123 Main St, Columbus, OH 43215')).toBeTruthy();
    expect(getByText('(555) 123-4567')).toBeTruthy();
  });

  it('should handle missing business data gracefully', () => {
    const incompleteBusiness: YelpBusiness = {
      id: 'incomplete-business',
      name: 'Incomplete Business',
      // Missing most optional fields
    };

    const { getByText, queryByTestId } = render(
      <TestHarness business={incompleteBusiness}>
        <BusinessCardModal />
      </TestHarness>
    );

    // Should show business name
    expect(getByText('Incomplete Business')).toBeTruthy();
    
    // Should not show missing optional fields
    expect(queryByTestId('bqi-rating')).toBeNull();
    expect(queryByTestId('bqi-reviews')).toBeNull();
    expect(queryByTestId('bqi-price')).toBeNull();
    expect(queryByTestId('bqi-categories')).toBeNull();
  });

  describe('snapshots', () => {
    it('should match snapshot for QuickInfo view', () => {
      const { toJSON } = render(
        <TestHarness>
          <BusinessCardModal />
        </TestHarness>
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot for Details view', () => {
      const { getByTestId, toJSON } = render(
        <TestHarness>
          <BusinessCardModal />
        </TestHarness>
      );

      // Switch to Details view
      fireEvent.press(getByTestId('details-tab'));

      expect(toJSON()).toMatchSnapshot();
    });
  });
});