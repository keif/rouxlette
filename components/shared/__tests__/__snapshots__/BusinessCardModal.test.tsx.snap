// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessCardModal snapshots should match snapshot for Details view 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
      }
    }
    testID="modal-backdrop"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "backgroundColor": "#fff",
              "borderRadius": 12,
              "elevation": 8,
              "maxHeight": "80%",
              "maxWidth": 400,
              "shadowColor": "#333",
              "shadowOffset": {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "borderBottomColor": "#99940",
                "borderBottomWidth": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "paddingHorizontal": 12,
                    "paddingVertical": 16,
                  },
                  false,
                ]
              }
              testID="quick-info-tab"
            >
              <Text
                style={
                  [
                    {
                      "color": "#999",
                      "fontFamily": "WorkSans-Medium",
                      "fontSize": 16,
                    },
                    false,
                  ]
                }
              >
                Quick Info
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "paddingHorizontal": 12,
                    "paddingVertical": 16,
                  },
                  {
                    "borderBottomColor": "#D4AF37",
                    "borderBottomWidth": 3,
                  },
                ]
              }
              testID="details-tab"
            >
              <Text
                style={
                  [
                    {
                      "color": "#999",
                      "fontFamily": "WorkSans-Medium",
                      "fontSize": 16,
                    },
                    {
                      "color": "#D4AF37",
                      "fontFamily": "WorkSans-Bold",
                    },
                  ]
                }
              >
                Details
              </Text>
            </View>
          </View>
          <RCTScrollView
            showsVerticalScrollIndicator={false}
            style={
              {
                "maxHeight": "100%",
              }
            }
          >
            <View>
              <View
                style={
                  {
                    "padding": 16,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#333",
                      "fontFamily": "WorkSans-Bold",
                      "fontSize": 24,
                      "marginBottom": 20,
                      "textAlign": "center",
                    }
                  }
                  testID="bd-title"
                >
                  Test Restaurant
                </Text>
                <View
                  style={
                    {
                      "marginBottom": 20,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontFamily": "WorkSans-Bold",
                        "fontSize": 18,
                        "marginBottom": 8,
                      }
                    }
                  >
                    Address
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontFamily": "WorkSans-Regular",
                        "fontSize": 16,
                        "lineHeight": 22,
                      }
                    }
                    testID="bd-address"
                  >
                    123 Main St, Columbus, OH 43215
                  </Text>
                </View>
                <View
                  style={
                    {
                      "marginBottom": 20,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontFamily": "WorkSans-Bold",
                        "fontSize": 18,
                        "marginBottom": 8,
                      }
                    }
                  >
                    Contact
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontFamily": "WorkSans-Regular",
                        "fontSize": 16,
                      }
                    }
                    testID="bd-phone"
                  >
                    (555) 123-4567
                  </Text>
                </View>
                <View
                  style={
                    {
                      "marginBottom": 20,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontFamily": "WorkSans-Bold",
                        "fontSize": 18,
                        "marginBottom": 8,
                      }
                    }
                  >
                    Hours
                  </Text>
                  <View
                    style={
                      {
                        "backgroundColor": "#fff",
                        "borderColor": "#99940",
                        "borderRadius": 8,
                        "borderWidth": 1,
                        "padding": 12,
                      }
                    }
                    testID="bd-hours"
                  >
                    <View
                      style={
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "paddingVertical": 4,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#333",
                            "fontFamily": "WorkSans-Medium",
                            "fontSize": 14,
                            "width": 80,
                          }
                        }
                      >
                        Mon
                        :
                      </Text>
                      <Text
                        style={
                          {
                            "color": "#999",
                            "flex": 1,
                            "fontFamily": "WorkSans-Regular",
                            "fontSize": 14,
                            "textAlign": "right",
                          }
                        }
                      >
                        9:00 AM–5:00 PM
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "paddingVertical": 4,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#333",
                            "fontFamily": "WorkSans-Medium",
                            "fontSize": 14,
                            "width": 80,
                          }
                        }
                      >
                        Tue
                        :
                      </Text>
                      <Text
                        style={
                          {
                            "color": "#999",
                            "flex": 1,
                            "fontFamily": "WorkSans-Regular",
                            "fontSize": 14,
                            "textAlign": "right",
                          }
                        }
                      >
                        Closed
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "paddingVertical": 4,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#333",
                            "fontFamily": "WorkSans-Medium",
                            "fontSize": 14,
                            "width": 80,
                          }
                        }
                      >
                        ...
                        :
                      </Text>
                      <Text
                        style={
                          {
                            "color": "#999",
                            "flex": 1,
                            "fontFamily": "WorkSans-Regular",
                            "fontSize": 14,
                            "textAlign": "right",
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "marginTop": 20,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "gap": 12,
                        "justifyContent": "space-around",
                        "marginBottom": 12,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#D4AF37",
                          "borderRadius": 8,
                          "flex": 1,
                          "paddingHorizontal": 16,
                          "paddingVertical": 12,
                        }
                      }
                      testID="bd-map-btn"
                    >
                      <Text
                        style={
                          {
                            "color": "#fff",
                            "fontFamily": "WorkSans-Bold",
                            "fontSize": 14,
                          }
                        }
                      >
                        📍 Map
                      </Text>
                    </View>
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#D4AF37",
                          "borderRadius": 8,
                          "flex": 1,
                          "paddingHorizontal": 16,
                          "paddingVertical": 12,
                        }
                      }
                      testID="bd-call-btn"
                    >
                      <Text
                        style={
                          {
                            "color": "#fff",
                            "fontFamily": "WorkSans-Bold",
                            "fontSize": 14,
                          }
                        }
                      >
                        📞 Call
                      </Text>
                    </View>
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#D4AF37",
                          "borderRadius": 8,
                          "flex": 1,
                          "paddingHorizontal": 16,
                          "paddingVertical": 12,
                        }
                      }
                      testID="bd-yelp-btn"
                    >
                      <Text
                        style={
                          {
                            "color": "#fff",
                            "fontFamily": "WorkSans-Bold",
                            "fontSize": 14,
                          }
                        }
                      >
                        🔗 Yelp
                      </Text>
                    </View>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#999",
                        "borderRadius": 8,
                        "paddingHorizontal": 16,
                        "paddingVertical": 12,
                      }
                    }
                    testID="bd-close-btn"
                  >
                    <Text
                      style={
                        {
                          "color": "#333",
                          "fontFamily": "WorkSans-Bold",
                          "fontSize": 16,
                        }
                      }
                    >
                      Close
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`BusinessCardModal snapshots should match snapshot for QuickInfo view 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
      }
    }
    testID="modal-backdrop"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            {
              "backgroundColor": "#fff",
              "borderRadius": 12,
              "elevation": 8,
              "maxHeight": "80%",
              "maxWidth": 400,
              "shadowColor": "#333",
              "shadowOffset": {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "borderBottomColor": "#99940",
                "borderBottomWidth": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "paddingHorizontal": 12,
                    "paddingVertical": 16,
                  },
                  {
                    "borderBottomColor": "#D4AF37",
                    "borderBottomWidth": 3,
                  },
                ]
              }
              testID="quick-info-tab"
            >
              <Text
                style={
                  [
                    {
                      "color": "#999",
                      "fontFamily": "WorkSans-Medium",
                      "fontSize": 16,
                    },
                    {
                      "color": "#D4AF37",
                      "fontFamily": "WorkSans-Bold",
                    },
                  ]
                }
              >
                Quick Info
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "paddingHorizontal": 12,
                    "paddingVertical": 16,
                  },
                  false,
                ]
              }
              testID="details-tab"
            >
              <Text
                style={
                  [
                    {
                      "color": "#999",
                      "fontFamily": "WorkSans-Medium",
                      "fontSize": 16,
                    },
                    false,
                  ]
                }
              >
                Details
              </Text>
            </View>
          </View>
          <RCTScrollView
            showsVerticalScrollIndicator={false}
            style={
              {
                "maxHeight": "100%",
              }
            }
          >
            <View>
              <View
                style={
                  {
                    "padding": 16,
                  }
                }
              >
                <Image
                  source={
                    {
                      "uri": "https://example.com/image.jpg",
                    }
                  }
                  style={
                    {
                      "borderRadius": 8,
                      "height": 200,
                      "marginBottom": 12,
                      "width": "100%",
                    }
                  }
                  testID="bqi-image"
                />
                <Text
                  style={
                    {
                      "color": "#333",
                      "fontFamily": "WorkSans-Bold",
                      "fontSize": 24,
                      "marginBottom": 8,
                    }
                  }
                  testID="bqi-title"
                >
                  Test Restaurant
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "marginBottom": 8,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#D4AF37",
                        "fontFamily": "WorkSans-Medium",
                        "fontSize": 16,
                        "marginRight": 8,
                      }
                    }
                    testID="bqi-rating"
                  >
                    4.5
                    ★
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontFamily": "WorkSans-Regular",
                        "fontSize": 14,
                        "marginRight": 8,
                      }
                    }
                    testID="bqi-reviews"
                  >
                    128
                     reviews
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#059669",
                        "fontFamily": "WorkSans-Medium",
                        "fontSize": 16,
                      }
                    }
                    testID="bqi-price"
                  >
                    $$
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#999",
                      "fontFamily": "WorkSans-Regular",
                      "fontSize": 14,
                      "marginBottom": 12,
                    }
                  }
                  testID="bqi-categories"
                >
                  Italian, Pizza
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginBottom": 8,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#333",
                        "flex": 1,
                        "fontFamily": "WorkSans-Medium",
                        "fontSize": 14,
                      }
                    }
                    testID="bqi-today"
                  >
                    Today: 
                    9:00 AM–5:00 PM
                  </Text>
                  <View
                    style={
                      [
                        {
                          "borderRadius": 12,
                          "paddingHorizontal": 8,
                          "paddingVertical": 4,
                        },
                        {
                          "backgroundColor": "#05966920",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "WorkSans-Bold",
                            "fontSize": 12,
                          },
                          {
                            "color": "#059669",
                          },
                        ]
                      }
                      testID="bqi-status"
                    >
                      Open
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    {
                      "color": "#999",
                      "fontFamily": "WorkSans-Regular",
                      "fontSize": 14,
                      "marginBottom": 16,
                    }
                  }
                  testID="bqi-distance"
                >
                  📍 
                  0.3 mi
                </Text>
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "gap": 12,
                      "justifyContent": "space-around",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#D4AF37",
                        "borderRadius": 8,
                        "flex": 1,
                        "paddingHorizontal": 16,
                        "paddingVertical": 12,
                      }
                    }
                    testID="bqi-details-btn"
                  >
                    <Text
                      style={
                        {
                          "color": "#fff",
                          "fontFamily": "WorkSans-Bold",
                          "fontSize": 16,
                        }
                      }
                    >
                      Details
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#D4AF37",
                        "borderRadius": 8,
                        "flex": 1,
                        "paddingHorizontal": 16,
                        "paddingVertical": 12,
                      }
                    }
                    testID="bqi-yelp-btn"
                  >
                    <Text
                      style={
                        {
                          "color": "#fff",
                          "fontFamily": "WorkSans-Bold",
                          "fontSize": 16,
                        }
                      }
                    >
                      Yelp
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                          "backgroundColor": "#D4AF37",
                          "borderRadius": 8,
                          "flex": 1,
                          "paddingHorizontal": 16,
                          "paddingVertical": 12,
                        },
                        {
                          "backgroundColor": "#999",
                        },
                      ]
                    }
                    testID="bqi-close-btn"
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "#fff",
                            "fontFamily": "WorkSans-Bold",
                            "fontSize": 16,
                          },
                          {
                            "color": "#333",
                          },
                        ]
                      }
                    >
                      Close
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
